### PR TITLE
API v2 changes

### LINKED ISSUE #2398: Re-root API at /api/core/v2
Re-root the REST API at /api/core/v2.

* Change routes in APId
* Update routes in sensuctl client

Refs #2252

### LINKED ISSUE #2399: Change RBAC path prefix to /api/core/v2
Move RBAC "api group" into `/api/core/v2` prefix.
ex: `/apis/rbac/v2/` would become `/api/core/v2/rbac`
- backend/apid/middlewares
- backend/apid/routers
- cli/client
- types (or api/core/v2 from #2397)

https://github.com/sensu/sensu-go/blob/cc846c027461b1eeb8dcad5454132175fb60b561/backend/apid/middlewares/authorization_attributes.go#L24-L28

https://github.com/sensu/sensu-go/blob/cc846c027461b1eeb8dcad5454132175fb60b561/backend/apid/routers/roles.go#L29

### LINKED ISSUE #2402: Make namespace part of the URL path
This task includes:

- Getting rid of the `namespace` query parameter
- Introducing the right `PathPrefix` for the route

Basically, the routes for all the resources are to be modeled after the routes for the RBAC resources.

Refs #2252

### PR DESCRIPTION
**Pull request by ccressent**
## What is this change?

* Clean up and standardize routers/controllers
* Make API more uniform by re-rooting it and having namespaces part of the URL
* Update CLI client accordingly

## Why is this change necessary?

Closes #2398 
Closes #2399 
Closes #2402 

## Does your change need a Changelog entry?

No.

## Were there any complications while making this change?

We unearthed an issue with RBAC, default permissions and listing namespaces. We've talked about it and have a plan to address the issue.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

No.

### CODE REVIEW

File: backend/apid/middlewares/authorization_attributes.go [hunk 1]
```diff
@@ -56,129 +55,24 @@ func (a AuthorizationAttributes) Then(next http.Handler) http.Handler {
...
 
 		// Most resource names are identified by a route variable named "id".
 		// Other resources have snowflake paths; see their corresponding router
 		// and the expected paths above.
-		attrs.ResourceName = vars["id"]
-
 		switch attrs.Resource {
 		case "events":
 			attrs.ResourceName = path.Join(vars["entity"], vars["check"])
 		case "silenced":
```
Comment (palourde) [inline backend/apid/middlewares/authorization_attributes.go]: Somehow listing silenced entries doesn't seem to work anymore:

```
$ sensuctl silenced create -s 'entity:whisky' --reason 'foo'
$ sensuctl silenced create -s 'entity:whisky' --check check-cpu --reason 'bar'
$ sensuctl silenced list
Error: not found
$ sensuctl silenced info -s 'entity:whisky'
=== entity:whisky:*
Expire:          -1s
ExpireOnResolve: false
Creator:         admin
Check:
Reason:          foo
Subscription:    entity:whisky
Namespace:       default
$ sensuctl silenced info 'entity:whisky:check-cpu'
=== entity:whisky:check-cpu
Expire:          -1s
ExpireOnResolve: false
Creator:         admin
Check:           check-cpu
Reason:          bar
Subscription:    entity:whisky
Namespace:       default
```

File: backend/apid/routers/entities.go [hunk 2]
```diff
@@ -42,6 +43,7 @@ func (r *EntitiesRouter) destroy(req *http.Request) (interface{}, error) {
 		return nil, err
 	}
 	err = r.controller.Destroy(req.Context(), id)
+	fmt.Println(err)
```
Comment (ccressent) [inline backend/apid/routers/entities.go]: I'm assuming this was for debugging; will I remove it @palourde?

**Comment by palourde** [comment]
Yes my bad

File: backend/apid/middlewares/authorization_attributes_test.go [hunk 3]
```diff
@@ -125,10 +125,10 @@ func TestAuthorizationAttributes(t *testing.T) {
 
 			// Prepare the router
 			router := mux.NewRouter()
-			router.PathPrefix("/apis/{group}/{version}/namespaces/{namespace}/{resource}/{name}").Handler(testHandler)
-			router.PathPrefix("/apis/{group}/{version}/namespaces/{namespace}/{resource}").Handler(testHandler)
-			router.PathPrefix("/apis/{group}/{version}/{resource}/{name}").Handler(testHandler)
-			router.PathPrefix("/apis/{group}/{version}/{resource}").Handler(testHandler)
+			router.PathPrefix("/api/{group}/{version}/namespaces/{namespace}/{resource}/{id}").Handler(testHandler)
```
[the remark this answers is not in the document]
Comment (ccressent) [reply backend/apid/middlewares/authorization_attributes_test.go]: I don't know what was agreed on, but the fact is that it was still `id` in all the routers. So I went with the path of least change and went with `id`!

**Comment by amdprophet** [comment]
Were the tests removed because they're all broken after recent changes? Do we want to add them back post-GA?

**Comment by ccressent** [comment]
@amdprophet: in a nutshell, they have been removed because they were actually not supposed to pass and only recent changes exposed that.

**Pull request merged**